### ISSUE TITLE
Can I build it in WIN

### ISSUE THREAD

**Issue by zhanghongyong123456** [tensorlayer/hyperpose#377]
I looked at several questions to try to use it in Win, but I didn't see if it was successful in Win, so I wanted to ask if it was possible to implement win build

**Comment by ganler** [comment]
We don't have official guidance to build it on windows. Though I knew that some users have built it on Windows since all of our codes are cross-platform. I would suggest you build it with WSL2 which will bring you a Linux environment so that you can simply follow the instructions from the tutorial.

**Comment by zhanghongyong123456** [comment]
> We don't have official guidance to build it on windows. Though I knew that some users have built it on Windows since all of our codes are cross-platform. I would suggest you build it with WSL2 which will bring you a Linux environment so that you can simply follow the instructions from the tutorial.

Thank you for your reply, how do I build locally on ubuntu 18.04 instead of using docker?

**Comment by ganler** [comment]
> > We don't have official guidance to build it on windows. Though I knew that some users have built it on Windows since all of our codes are cross-platform. I would suggest you build it with WSL2 which will bring you a Linux environment so that you can simply follow the instructions from the tutorial.
> 
> Thank you for your reply, how do I build locally on ubuntu 18.04 instead of using docker?

https://hyperpose.readthedocs.io/en/latest/markdown/install/prediction.html#build-from-source

**Comment by zhanghongyong123456** [comment]
> > > We don't have official guidance to build it on windows. Though I knew that some users have built it on Windows since all of our codes are cross-platform. I would suggest you build it with WSL2 which will bring you a Linux environment so that you can simply follow the instructions from the tutorial.
> > 
> > 
> > Thank you for your reply, how do I build locally on ubuntu 18.04 instead of using docker?
> 
> https://hyperpose.readthedocs.io/en/latest/markdown/install/prediction.html#build-from-source

Thank you for your prompt reply

**Comment by zhanghongyong123456** [comment]
> > > > We don't have official guidance to build it on windows. Though I knew that some users have built it on Windows since all of our codes are cross-platform. I would suggest you build it with WSL2 which will bring you a Linux environment so that you can simply follow the instructions from the tutorial.
> > > 
> > > 
> > > Thank you for your reply, how do I build locally on ubuntu 18.04 instead of using docker?
> > 
> > 
> > https://hyperpose.readthedocs.io/en/latest/markdown/install/prediction.html#build-from-source
> 
> Thank you for your prompt reply

Yes, I built him on ubuntu18.04. Able to realize real-time processing of video streams，like this： ./hyperpose-cli --runtime=operator --source=../data/media/video.avi ，
Q1 ：I want to modify the output format, what should I do, like openpose's output format json file (25 key points)
Q2：Is our key point detection consistent with the key point of openpose body 25 keypoint or 18coco keypoint?
![image](https://user-images.githubusercontent.com/48466610/128683825-602eeb3b-6a57-44d9-9cfb-a5be41c82057.png)
![image](https://user-images.githubusercontent.com/48466610/128683890-d5ac7b4e-990d-4101-915c-020ade827cad.png)

**Comment by ganler** [comment]
we do 18 coco keypoints. Notably, when supporting openpifpaf, we convert its format to approximate 18 coco format. As for extending a Json output, you can modify the source codes in stream.(hpp or cpp) to add a new output operator.

**Comment by zhanghongyong123456** [comment]
> we do 18 coco keypoints. Notably, when supporting openpifpaf, we convert its format to approximate 18 coco format. As for extending a Json output, you can modify the source codes in stream.(hpp or cpp) to add a new output operator.

I tried it, but it didn't work，I would like to ask where to get the key point code, I am not very familiar with the code from C++, can you have detailed information and implementation examples，Looking forward to your reply

**Comment by ganler** [comment]
@zhanghongyong123456 See https://github.com/tensorlayer/hyperpose/blob/master/src/stream.cpp#L133

I would appreciate it if you can contribute a JSON writer.
